python/lsst/afw/geom/span.cc

```diff
@@ -0,0 +1,95 @@
+/* 
+ * LSST Data Management System
+ * Copyright 2008-2016  AURA/LSST.
+ * 
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the LSST License Statement and 
+ * the GNU General Public License along with this program.  If not, 
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include <pybind11/pybind11.h>
+#include <pybind11/operators.h>
+
+#include "lsst/afw/geom/Span.h"
+#include "lsst/afw/geom/SpanPixelIterator.h"
+
+namespace py = pybind11;
+
+using namespace lsst::afw::geom;
+
+// A thin wrapper around SpanPixelIterator.
+// Unfortunately we cannot use py::make_iterator here, as we normally
+// should, because for some reason the return values then all refer
```

Comment:
I don't think this is the problem. At least pybind11 doesn't seem to know about iterator categories. It must somehow be a problem with our boost-iterator-facade using implementation and pybind11 expecting a standard STL iterator. It must behave differently somehow (although I also think it really shouldn't). But I'm not sure. Also not sure if we really want to dig in deep here (now at least). I suggest we reevaluate when we have stripped out Boost.